src/pulselistener/pulselistener/lib/phabricator.py

```diff
@@ -0,0 +1,201 @@
+
+        # Load secure projects
+        projects = self.api.search_projects(slugs=['secure-revision'])
+        self.secure_projects = {
+            p['phid']: p['fields']['name']
+            for p in projects
+        }
+        logger.info('Loaded secure projects', projects=self.secure_projects.values())
+
+    def update_state(self, build):
+        '''
+        Check the visibility of the revision, by retrying N times with a specified time
+        This method is executed regularly by the client application to check on the status evolution
+        as the BMO daemon can take several minutes to update the status
+        '''
+        assert isinstance(build, PhabricatorBuild)
+
+        # Only when queued
+        if build.state != PhabricatorBuildState.Queued:
+            return
+
+        # Check this build has some retries left
+        retries_left, last_try = self.retries[build.target_phid]
+        if retries_left <= 0:
+            return
+
+        # Check this build has been awaited between tries
+        now = time.time()
+        if last_try is not None and now - last_try < self.sleep:
+            return
+
+        # Now we can check if this revision is public
+        retries_left -= 1
+        self.retries[build.target_phid] = (retries_left, now)
+        logger.info('Checking visibility status', build=str(build), retries_left=retries_left)
+
+        if self.is_visible(build):
+            build.state = PhabricatorBuildState.Public
+            logger.info('Revision is public', build=str(build))
+
+        elif retries_left <= 0:
+            # Mark as secured when no retries are left
+            build.state = PhabricatorBuildState.Secured
+            logger.info('Revision is marked as secure', build=str(build))
+
+        else:
+            # Enqueue back to retry later
+            build.state = PhabricatorBuildState.Queued
+
+    def is_visible(self, build):
+        '''
+        Check the visibility of the revision by loading its details
+        '''
+        assert isinstance(build, PhabricatorBuild)
+        assert build.state == PhabricatorBuildState.Queued
+        try:
+            # Load revision with projects
+            build.revision = self.api.load_revision(
+                rev_id=build.revision_id,
+                attachments={'projects': True, 'reviewers': True}
+            )
+            if not build.revision:
+                raise Exception('Not found')
+
+            # Check against secure projects
+            projects = set(build.revision['attachments']['projects']['projectPHIDs'])
+            if projects.intersection(self.secure_projects):
+                raise Exception('Secure revision')
+        except Exception as e:
+            logger.info('Revision not accessible', build=str(build), error=str(e))
+            return False
+
+        return True
+
+    def load_patches_stack(self, build):
```

Comment:
I'd keep this into libmozdata, and remove the `revision_available` stuff from there (or make it optional) so you are not forced to use it.